cmd/statusd/main.go

```diff
@@ -184,3 +205,25 @@ Options:
 	fmt.Fprintf(os.Stderr, usage) // nolint: gas
 	flag.PrintDefaults()
 }
+
+func haltOnInterruptSignal(nodeManager common.NodeManager) {
+	signalCh := make(chan os.Signal, 1)
+	signal.Notify(signalCh, os.Interrupt)
+	defer signal.Stop(signalCh)
+	<-signalCh
+
+	log.Println("Got interrupt, shutting down...")
+
+	nodeStopped, err := nodeManager.StopNode()
```

Comment:
Don't we have a synchronous `StopNode()`? Found this construct pretty often. IMHO concurrency in cases like here should be only be added by the caller if needed. Maybe something to discuss for a refactoring.

Reply:
Exactly! We don't have a synchronous method and this is what I would like to fix in https://github.com/status-im/ideas/issues/63. `NodeManager` should not implement these methods as async at all.

Reply:
Top, already like idea #63 right now. 😄